test/gurka/test_use.cc

```diff
@@ -0,0 +1,203 @@
+                                {"amenity", "yes"},
+                                {"destination", "Service Area"}}},
+                              {"EG", {{"highway", "motorway"}}}};
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize_metres);
+    map = gurka::buildtiles(layout, ways, {}, {}, "test/data/gurka_use", build_config);
+  }
+};
+gurka::map Use::map = {};
+
+/*************************************************************/
+
+TEST_F(Use, EdgeUse) {
+  auto result = gurka::route(map, "A", "C", "auto");
+
+  // rest_area
+  ASSERT_EQ(result.trip().routes(0).legs_size(), 1);
+  auto leg = result.trip().routes(0).legs(0);
+  EXPECT_EQ(leg.node(1).edge().use(), TripLeg::Use::TripLeg_Use_kRestAreaUse); // BC
+
+  // service_area
+  result = gurka::route(map, "A", "F", "auto");
+
+  leg = result.trip().routes(0).legs(0);
+  EXPECT_EQ(leg.node(3).edge().use(), TripLeg::Use::TripLeg_Use_kServiceAreaUse); // EF
+}
+
+TEST_F(Use, test_passing_rest_area) {
+  auto result = gurka::route(map, "A", "D", "auto");
+  auto d = gurka::convert_to_json(result, valhalla::Options::Format::Options_Format_osrm);
+
+  // Assert expected number of routes, legs, steps
+  ASSERT_EQ(d["routes"].Size(), 1);
+  ASSERT_EQ(d["routes"][0]["legs"].Size(), 1);
+  ASSERT_EQ(d["routes"][0]["legs"][0]["steps"].Size(), 2);
+  auto steps = d["routes"][0]["legs"][0]["steps"].GetArray();
+
+  // Expect maneuvers
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kDestination});
+
+  // Expect only the second intersection of the first step to have rest_stops
+  // Expect rest_stops to be of type=rest_area
+  int step_idx = 0;
+  EXPECT_EQ(steps[step_idx]["intersections"].Size(), 2);
+  EXPECT_FALSE(steps[step_idx]["intersections"][0].HasMember("rest_stops"));
+  EXPECT_TRUE(steps[step_idx]["intersections"][1].HasMember("rest_stops"));
+  EXPECT_EQ(steps[step_idx]["intersections"][1]["rest_stops"].Size(), 1);
+  EXPECT_STREQ(steps[step_idx]["intersections"][1]["rest_stops"][0]["type"].GetString(), "rest_area");
+
+  // Expect no rest stops on last step
+  step_idx++;
+  EXPECT_EQ(steps[step_idx]["intersections"].Size(), 1);
+  EXPECT_FALSE(steps[step_idx]["intersections"][0].HasMember("rest_stops"));
+}
+
+TEST_F(Use, test_entering_rest_area) {
+  auto result = gurka::route(map, "A", "C", "auto");
+  auto d = gurka::convert_to_json(result, valhalla::Options::Format::Options_Format_osrm);
+
+  // Assert expected number of routes, legs, steps
+  ASSERT_EQ(d["routes"].Size(), 1);
+  ASSERT_EQ(d["routes"][0]["legs"].Size(), 1);
+  ASSERT_EQ(d["routes"][0]["legs"][0]["steps"].Size(), 3);
+  auto steps = d["routes"][0]["legs"][0]["steps"].GetArray();
+
+  // Expect maneuvers
+  gurka::assert::raw::expect_maneuvers(result, {DirectionsLeg_Maneuver_Type_kStart,
+                                                DirectionsLeg_Maneuver_Type_kLeft,
```

Comment:
@dgearhart I think this shows a problem with guidance when the ramp has a rest area use -- we call out a Turn Left instead of an Exit Left. Can we use this test to figure out what changes we need to make in Odin?

Reply:
@danpaz yes, I already [commented](https://github.com/mapbox/valhalla-data/issues/215#issuecomment-690370405) that guidance code will need to be updated to work with the new `Use` values